### ISSUE TITLE
Implement `Interpolate` for basic `bevy` types

### ISSUE THREAD

**Issue by Umatriz** [Bendzae/bevy_replicon_snap#6]
I think there should be an implementation of `Interpolate` trait for basic `bevy` types. The most obvious example is `Transform`. It might be behing a feature flag so people will turn it on if needed.

**Comment by Bendzae** [comment]
True, it would make sense for the basic bevy types. If you have time, feel free to create a PR for that. Otherwise I might add it in the future.

**Comment by olukowski** [comment]
I think I will go ahead and implement this, I need it for my own project anyway.

**Comment by olukowski** [comment]
For now I will just do transform, I will open a PR so if I need to change anything, let me know.

**Comment by Darkluke1111** [comment]
Is a feature flag even necessary for this? I feel like it would be opt-in even without a flag since you still need to call `app.replicate_interpolated::<Transform>()` to use the interpolation.
